google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -482,6 +489,10 @@ private void doneCallback(Throwable finalStatus) {
     } finally {
       this.lock.unlock();
     }
+    Exceptions.StorageException storageException = Exceptions.toStorageException(finalStatus);
+    if (storageException != null) {
+      throw new StorageException(finalStatus);
```

Comment:
Just set the connectionFinalStatus to be the upgraded stroageException like before.

Reply:
Done